filters/openpolicyagent/tracing.go

```diff
@@ -14,14 +20,43 @@ func init() {
 type tracingFactory struct{}
 
 type transport struct {
-	opa     *OpenPolicyAgentInstance
+	tracer     opentracing.Tracer
+	bundleName string
+	manager    *plugins.Manager
+
 	wrapped http.RoundTripper
 }
 
 func (*tracingFactory) NewTransport(tr http.RoundTripper, opts opatracing.Options) http.RoundTripper {
+	var tracer opentracing.Tracer
+
+	if len(opts) < 3 {
+		panic("insufficient tracing options provided for outbound http calls from Open Policy Agent")
+	}
+
+	if opts[0] != nil {
+		var ok bool
+		tracer, ok = opts[0].(opentracing.Tracer)
+		if !ok {
+			panic("invalid type for first argument of tracing options, expected opentracing.Tracer")
+		}
+	}
+
+	bundleName, ok := opts[1].(string)
+	if !ok {
+		panic("invalid type for second argument of tracing options, expected string")
+	}
+
+	manager, ok := opts[2].(*plugins.Manager)
+	if !ok {
+		panic("invalid type third argument of tracing options, expected *plugins.Manager")
+	}
```

Comment:
This imposes an order on the options which is error-prone.
One way is to process them in loop (and for string arg define typed alias `type BundleName string`):
```go
for _, o := range opts {
   switch o := o.(type) {
    ...
   }
}
```

Another way is to use functional options, see https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
I guess we can also change signature and supply Options struct

Reply:
Changed this to functional options as this fits with the rest of the filters which also use the same pattern...